packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -1301,6 +1322,55 @@ export const useArbTokenBridge = (
     setExecutedMessagesCache({ ...executedMessagesCache, ...added })
   }
 
+  const tokensFromLists: SearchableTokenStorage = useMemo(() => {
+    const l1Network = l1.network
+    const l2Network = l2.network
+
+    const tokenLists = getTokenLists(String(l2Network.chainID))
+
+    if (typeof l1Network === 'undefined' || typeof l2Network === 'undefined') {
+      return {}
+    }
+
+    return tokenListsToSearchableTokenStorage(
+      tokenLists,
+      String(l1Network.chainID),
+      String(l2Network.chainID)
+    )
+  }, [l1.network, l2.network])
+
+  const tokensFromUser: SearchableTokenStorage = useMemo(() => {
+    const storage: SearchableTokenStorage = {}
+
+    // Can happen when switching networks.
+    if (typeof bridgeTokens === 'undefined') {
+      return {}
+    }
+
+    Object.keys(bridgeTokens).forEach((_address: string) => {
+      const bridgeToken = bridgeTokens[_address]
+
+      // Any tokens in the bridge that don't have a list id were added by the user.
+      if (bridgeToken && !bridgeToken.listID) {
+        storage[_address] = { ...bridgeToken, tokenLists: [] }
+      }
+    })
+    return storage
+  }, [bridgeTokens])
+
+  /*
+    function to search for a token within the union of token lists (pre-configured + user added)
+    Objective : if basic data like { symbol, name, decimals } is reqd, then why query the chain for the token data already fetched.
+  */
+  const searchTokenFromList = useCallback(
```

Comment:
Can we use `bridgeTokens[tokenAddress]` instead? All tokens should be added to `bridgeTokens`